# Cálculo - Tarefa 02

## 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  

In [4]:
import numpy as np

def f(x):
    return x**3

a = 1
b = 2


n1 = 4
delta_x1 = (b - a) / n1
x_vals1 = np.linspace(a, b - delta_x1, n1)
f_vals1 = f(x_vals1)
area_4 = np.sum(f_vals1) * delta_x1

n2 = 100
delta_x2 = (b - a) / n2
x_vals2 = np.linspace(a, b - delta_x2, n2)
f_vals2 = f(x_vals2)
area_100 = np.sum(f_vals2) * delta_x2


print(f"Aproximação numérica com {n1} retângulos: {area_4}")
print(f"Aproximação numérica com {n2} retângulos: {area_100}")


Aproximação numérica com 4 retângulos: 2.921875
Aproximação numérica com 100 retângulos: 3.715075


## 2- Calcule essa área de forma analítica.

In [6]:
area_analitica = (b**4)/4 - (a**4)/4

print(f"Integral analítica exata: {area_analitica}")

Integral analítica exata: 3.75


$$
\int_1^2 x^3 \, dx = \left[ \frac{x^4}{4} \right]_1^2 = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \boxed{3.75}
$$
